app/scripts/controllers/app-state-controller.ts

```diff
@@ -196,79 +202,239 @@ const getDefaultAppStateControllerState = (
   currentExtensionPopupId: 0,
 });
 
-export class AppStateController extends EventEmitter {
-  private readonly extension: AppStateControllerOptions['extension'];
+const controllerMetadata = {
+  timeoutMinutes: {
+    persist: true,
+    anonymous: true,
+  },
+  connectedStatusPopoverHasBeenShown: {
+    persist: true,
+    anonymous: true,
+  },
+  defaultHomeActiveTabName: {
+    persist: true,
+    anonymous: true,
+  },
+  browserEnvironment: {
+    persist: true,
+    anonymous: true,
+  },
+  popupGasPollTokens: {
+    persist: true,
+    anonymous: true,
+  },
+  notificationGasPollTokens: {
+    persist: true,
+    anonymous: true,
+  },
+  fullScreenGasPollTokens: {
+    persist: true,
+    anonymous: true,
+  },
+  recoveryPhraseReminderHasBeenShown: {
+    persist: true,
+    anonymous: true,
+  },
+  recoveryPhraseReminderLastShown: {
+    persist: true,
+    anonymous: true,
+  },
+  outdatedBrowserWarningLastShown: {
+    persist: true,
+    anonymous: true,
+  },
+  nftsDetectionNoticeDismissed: {
+    persist: true,
+    anonymous: true,
+  },
+  showTestnetMessageInDropdown: {
+    persist: true,
+    anonymous: true,
+  },
+  showBetaHeader: {
+    persist: true,
+    anonymous: true,
+  },
+  showPermissionsTour: {
+    persist: true,
+    anonymous: true,
+  },
+  showNetworkBanner: {
+    persist: true,
+    anonymous: true,
+  },
+  showAccountBanner: {
+    persist: true,
+    anonymous: true,
+  },
+  trezorModel: {
+    persist: true,
+    anonymous: true,
+  },
+  currentPopupId: {
+    persist: false,
+    anonymous: true,
+  },
+  onboardingDate: {
+    persist: false,
```

Comment:
We definitely want to track when the user was onboarded. If we lost that state we wouldn't be able to get it back.

```suggestion
    persist: true,
```